src/inet/UDPEndPointImplSockets.cpp

```diff
@@ -93,6 +93,8 @@ namespace {
 
 CHIP_ERROR IPv6Bind(int socket, const IPAddress & address, uint16_t port, InterfaceId interface)
 {
+    VerifyOrReturnValue(socket >= 0, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Again, this is only called if GetSocket succeeds, which guarantees that socket >= 0, no?

Does it help if GetSocket() checks for < 0, instead of just checking for == -1, after doing `::socket()`?

If not, can we restructure this stuff somehow (e.g. have the socket be an `optional<unsigned>` instead of `int` so that it's crystal-clear that if we have a value it is in fact not negative?).

Reply:
negative check does not help: tidy knows it is exactly -1.
What clang-tidy says that "if we return a CHIP_ERROR_POSIX(errno) and IsSuccess returns true on that, then error occurs". However this is incompatible since CHIP_ERROR will not be a success, it is just clang-tidy assumes IsSuccess probably can change over time and cannot infer the CHIP_ERROR_POSIX is never a success in our usage.

Reply:
It would be fixable if GetSocket would return some `variant<CHIP_ERROR, unsigned>` and we assign that to mSocket. But that starts being odd ... we are re-writing the code a lot.